src/arch/x86_64/mod.rs

```diff
@@ -43,7 +43,7 @@ pub fn detect_freq_from_cpuid(
 	let tsc_frequency_hz = cpuid.get_tsc_info().map(|tinfo| {
 		if tinfo.tsc_frequency().is_some() {
 			tinfo.tsc_frequency()
-		} else {
+		} else if tinfo.denominator() != 0 && tinfo.numerator() != 0 {
```

Comment:
According to the [docs](https://docs.rs/raw-cpuid/11.0.2/raw_cpuid/struct.TscInfo.html#method.numerator), we can as well just check for the numerator:

```suggestion
		} else if tinfo.numerator() != 0 {
```